src/TurboMqtt/Streams/ClientAckingFlow.cs

```diff
@@ -72,89 +75,88 @@ private sealed class Logic : InAndOutGraphStageLogic
         public Logic(ClientAckingFlow stage) : base(stage.Shape)
         {
             _stage = stage;
-            _publishIds = new SimpleLruCache<NonZeroUInt16>(stage._bufferSize, stage._bufferExpiry);
-            _pubRelIds = new SimpleLruCache<NonZeroUInt16>(stage._bufferSize, stage._bufferExpiry);
```

Comment:
Profiled this - the explicit de-duplication code was actually a major source of slow-downs. Think O(N) for each message that came in. I'm rethinking whether or not this is the client's responsibility or not too - we should add the `PacketId` to the `MqttMessage` type if we want to pass the buck on de-duplication to the end-user.